primer/react
# Adapts Blankslate to render proportionally in narrow areas

Relates to https://github.com/github/primer/issues/2678

**Before:**
![Screenshot 2023-10-26 at 2 15 42 PM](https://github.com/primer/react/assets/2313998/18cda6e8-ca9d-4525-8e26-3aa5e88e7817)

**After:**
![Screenshot 2023-10-26 at 2 16 03 PM](https://github.com/primer/react/assets/2313998/ed24b9cb-48c9-4f4b-ac32-1e8f5e68ac7a)

### Changelog

#### New

Nothing new

#### Changed

- Upgrades `styled-components` so we can use the `@container` rule block
- Adds container queries to adapt the sizing and spacing based on the available area
- Uses Primer Primitives instead of relying on browser-native styles for text elements

#### Removed

Nothing removed

### Rollout strategy




### Testing & Reviewing

- Open any Blankslate story and resize the viewport so the Blankslate is < 545px
- Confirm that the sizing and spacing shrinks
- Confirm that no colors have changed

### Merge checklist


Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.
